ciao_contrib/_tools/specextract.py

```diff
@@ -909,6 +909,17 @@ def _check_merged_input(self,kwdict,toolname):
 Merged events files should not be used for spectral analysis.") # or v1 warning?
 
 
+    def _check_ppr(self,kwdict,toolname):
+        """
+        check if the input file has been generated by psf_project_ray
+        """
+        keys_to_check = [key for key in kwdict.keys() if key.upper().startswith("HDUCLAS")]
+
+        for pkey in keys_to_check:
+            if kwdict["HDUNAME"].lower() == "rayevent" and kwdict[pkey].lower() == "psfray":
+                raise IOError(f"Event files generated by psf_project_ray are not supported by {toolname}.")
+
+
```

Comment:
So, if I'm reading this correctly then we cam have

```
def _check_ppr(sef, kwdict, toolname):
    if kwdict["HDUNAME"].lower() != "rayevent":
        return

    keys_to_check = [...]
    for pkey in keys_to_check:
        if kwdict[pkey].lower() == "psfray":
            raise IOError(...)
```

and actually we can just combine the two loops - ie don't create `keys_to_check` so we just have

````
    for key, val in kwdict.items():
        if not key.upper().startswith("HDUCLAS") or val.lower() != "psfray":
            continue
        raise IOError(...)
```

Reply:
I've updated the function and tested it out... looks good